frontend/src/views/RoomSelectView.vue

```diff
@@ -21,20 +36,65 @@ import { useCurrentRoomStore } from "@/store";
 const roomData = ref<Room[]>([]);
 const router = useRouter();
 const currentRoomStore = useCurrentRoomStore();
+const isLoading = ref(false);
+const retryCount = ref(0);
+const maxRetries = 3;
 
 async function fetchRooms() {
+  if (isLoading.value) return;
+  
+  isLoading.value = true;
   try {
     const response = await apiClient.getRooms();
     if (response.success) {
       // ルームIDの昇順でソート
       roomData.value = response.data.sort((a: Room, b: Room) => a.roomId - b.roomId);
-      console.log(response.data);
+      console.log("Successfully fetched rooms:", response.data);
+      retryCount.value = 0; // 成功したらリトライカウントをリセット
     } else {
       console.error("Failed to fetch rooms:", response.data);
+      await handleFetchError(response.status);
     }
   } catch (error) {
     console.error("Error fetching rooms:", error);
+    await handleFetchError(0);
+  } finally {
+    isLoading.value = false;
+  }
+}
+
+async function handleFetchError(status: number) {
+  // 認証エラー（401）またはユーザーが見つからない（404）の場合はユーザー選択画面に戻る
+  if (status === 401 || status === 404) {
+    console.warn("Authentication failed, redirecting to signup");
+    alert("認証に失敗しました。ユーザー選択画面に戻ります。");
+    router.push("/");
+    return;
+  }
+
+  // その他のエラーでリトライ可能な場合は自動で再試行
+  if (retryCount.value < maxRetries) {
```

Comment:
```suggestion
  if (retryCount.value <= maxRetries) {
```
maxRetries を「再接続の試行回数上限」という意味を持つ変数と見る場合、3 回目の再接続の試み（最初を含めると 4 回目の接続の試み）も行ってほしいかも